app/services/user_service.py

```diff
@@ -0,0 +1,151 @@
+from fastapi import status
+from sqlalchemy.sql.expression import delete, insert, select
+
+from app import db
+from app.models import FriendRequests, Friends, User
+from app.schemas import UserResponse
+
+
+class UserService:
+    @classmethod
+    async def get_user_by_id(cls, id):
+        user = await db.fetch_one(select([User]).where(User.id == id))
+        if user is None:
+            return {
+                "message": "User not found",
+                "status_code": status.HTTP_404_NOT_FOUND,
+            }
+
+        return {
+            "message": "User found",
+            "data": UserResponse(**user),
+            "status_code": status.HTTP_200_OK,
+        }
+
+    @classmethod
+    async def get_user_by_name(cls, name):
+        user = await db.fetch_one(select([User]).where(User.username == name))
+        if user is None:
+            return {
+                "message": "User not found",
+                "status_code": status.HTTP_404_NOT_FOUND,
+            }
+        return {
+            "message": "User found",
+            "data": UserResponse(**user),
+            "status_code": status.HTTP_200_OK,
+        }
+
+    @classmethod
+    async def send_friend_request(cls, id, user):
+        friend = await cls.get_user_by_id(id)
+        if friend["status_code"] == status.HTTP_404_NOT_FOUND:
+            return friend
+
+        friend_id = friend.id
```

Comment:
`get_user_by_id` returns {"message": , "data": , "status_code"}, how did you access id from it. Are you sure this is working? no key errors?